nx_cugraph/algorithms/link_prediction.py

```diff
@@ -0,0 +1,99 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import cupy as cp
+import networkx as nx
+import pylibcugraph as plc
+
+from nx_cugraph.convert import _to_undirected_graph
+from nx_cugraph.utils import networkx_algorithm, not_implemented_for
+
+__all__ = [
+    "jaccard_coefficient",
+]
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@networkx_algorithm(version_added="25.02", _plc="jaccard_coefficients")
+def jaccard_coefficient(G, ebunch=None):
+    if ebunch is None:
+        # FIXME: is there a more efficient way to do this (on GPU or
+        # otherwise)?
+        ebunch = list(nx.non_edges(G))
+        if not ebunch:
+            return iter([])
+
+    G = _to_undirected_graph(G)
+
+    # FIXME: this zip() call appears to be the performance bottleneck for this
+    # function. Is there a better way?
+    (u, v) = zip(*ebunch)
+
+    try:
+        # Convert the ebunch lists to cupy arrays for passing to PLC, possibly
+        # mapping to integers if the Graph was renumbered.
+        # Allow the Graph renumber lookup (if renumbering was done) to check
+        # for invalid node IDs in ebunch.
+        u = G._list_to_nodearray(u)
+        v = G._list_to_nodearray(v)
+    except KeyError as n:
+        raise nx.NodeNotFound(f"Node {n} not in G.")
+    else:
+        # If G was not renumbered, then the ebunch nodes must be explicitly
+        # checked (note: ebunch can be very large). If not done,
+        # plc.jaccard_coefficients() will accept node IDs not in the graph and
+        # return a coefficient of 0 for them, which is not compatible with NX.
+        #
+        # FIXME: Is there a better way to do this?  Is there a utility to check
+        # if a node ID is valid for the graph?
+        if not hasattr(G, "key_to_id") or G.key_to_id is None:
+            ebunch_nodes = cp.unique(cp.concatenate([u, v]))
+            graph_nodes = cp.unique(
+                cp.concatenate(
+                    [
+                        G.src_indices,
+                        G.dst_indices,
+                        G._node_ids if G._node_ids is not None else cp.ndarray(0),
+                    ]
+                )
+            )
+            invalid = cp.setdiff1d(ebunch_nodes, graph_nodes, assume_unique=True)
+            if len(invalid) > 0:
+                raise nx.NodeNotFound(f"Node {invalid.tolist()[0]} not in G.")
+
+    # Note that Jaccard similarity must run on a symmetric graph.
+    # FIXME: PLC will symmetrize the graph if told to, but the symmetrize flag
+    # to _get_plc_graph() appears to symmetrize using cupy and does other
+    # things (cast to 64bit, etc.). Can we let PLC do the symmetrization if the
+    # symmetrize flag is set instead?
```

Comment:
> Can we let PLC do the symmetrization if the symmetrize flag is set instead?

Quite possibly. It should be straightforward to experiment now since this ought to be covered by tests.

I don't think this comment belongs here. It reads as either a comment that belongs in a PR discussion, or if we want a bread crumb for us to find later in the code, it should be findable in `_get_plc_graph`. Can you clean up or tighten some of your comments so that they reflect what is most helpful to future readers and perhaps add some comments to PR?